feat(Compass): add handbook doc page (#4877)
patternfly/patternfly-org: packages/documentation-site/patternfly-docs/content/AI/Compass/CompassHandbook.md

---
id: Compass
section: AI
source: guide
subsection: Generative UIs
---

When building a layout for generative UIs using the Compass components, there are several common patterns that may be implemented using Patternfly components and component variants.

> ### Compass structure

Reviewer: ```suggestion
## Structural patterns
```
unless you've specifically chosen these heading levels on purpose, just moving everything up a level

> #### Containers & glass theme

Reviewer: ```suggestion
### Transparent containers
```
or "Containers with the glass effect"

> The glass look and feel is an opt-in global styling applied when `pf-v6-theme-glass` is present on the `html` element class list, similar to dark theme. For compass layouts, content that has the glass background and rounded borders should be wrapped in a `CompassPanel` (and some compass components will do this automatically). This component has various options to customize its own padding and behavior to fit the context of its usage, and so is present in nearly every area of a compass page. `CompassPanels` should not be directly nested if the glass theme is applied to prevent styling issues.

Reviewer: ```suggestion
Transparent containers create a modern glass look and feel. To enable transparency on a container, apply `pf-v6-theme-glass` to the `html` element class list, similar to the approach for enabling the dark mode theme. 

Within Compass layouts, elements with a transparent glass background and rounded borders should be wrapped in a `<CompassPanel>` (some Compass components do this automatically). You can adjust various options of the `<CompassPanel>` to adapt its padding and behavior to fit the context of your use case. 

To prevent styling issues, do not nest`<CompassPanel>` components while using the glass effects.
```
wasn't sure what was meant by "and so is present in nearly every area of a compass page", so I just removed mention of it, but we can reintegrate it. What does this mean? That CompassPanels will be used a bunch on a page?

#### Header
A compass layout header will typically be constructed using a `CompassHeader` component, which constructs three sections: logo, nav, and profile. These sections may be passed any content, but here are some recommendations when building a compass layout:
- The logo should contain a `Brand` or other logo image or svg. 
- The profile should contain a `Dropdown` with a user `Avatar` within the `MenuToggle`.
- The nav should contain a `CompassNavContent` component, which usually contains `CompassNavHome`, `CompassNavMain`, and `CompassNavSearch` components as shown below in a snippet. `CompassNavHome` and `CompassNavSearch` are opinionated components that will automatically provide a home and search button, and are optional. `CompassNavMain` will usually contain a `Tabs` component with the `isNav` flag to enable a top-level navigation type of styling. 

```
<CompassNavContent>
  <CompassNavHome onClick={() => console.log('Home')} />
  <CompassNavMain>
    <Tabs isNav />
  </CompassNavMain>
  <CompassNavSearch onClick={() => console.log('Search')} />
</CompassNavContent>
```

> #### Sidebars

Reviewer: ```suggestion
### Sidebars
```

There are two sidebars in a compass layout at the start and end of the page. There is no specific `Compass` helper component for these sections, and so usually a `CompassPanel` containing an `ActionList` with the `isVertical` flag will be passed to represent the side navigations.

#### Footer
There are two methods of adding a footer to a compass layout. In either case, a footer's content will remain the same and will usually consist of a `@patternfly/chatbot` `MessageBar`, wrapped by a `CompassPanel` and `CompassMessageBar` component, as shown below:

> 

Reviewer: ```suggestion
There are 2 methods of adding a footer to a compass layout. 

In both methods, the footer content will remain the same, typically containing a [ChatBot `<MessageBar>`](/extensions/chatbot/messages/), wrapped in a `<CompassPanel>` and `<CompassMessageBar>` component:

```

```
 <CompassMessageBar>
  <CompassPanel isPill hasNoPadding hasNoBorder>
    <MessageBar />
    <div aria-live="polite" className="pf-v6-screen-reader">
      // aria anouncments for message bar's state updates
    </div>
  </CompassPanel>
</CompassMessageBar>
```

In the first method of adding a footer, pass this content to the `footer` prop of the `Compass` component. This type of footer will extend the entire width of the viewport. However, if the footer content is expected to change in height (such as when MessageBar content expands), the sidebars will adjust their positioning. If this behavior is not desired, please use the second method and forego passing the `footer` prop.

For the second method of adding a footer, wrap the content in `CompassMainFooter` and pass it as part of the `main` content of the `Compass`. This will allow the sidebars to extend through to the bottom of the viewport and prevents them from resizing based on changing footer content height.

> #### Main content

Reviewer: ```suggestion
### Main content
```

---
id: Compass
section: AI
source: Development-guide
subsection: Generative UIs
---

When building generative UI layouts using the Compass components, there are several common patterns to considering during your development.

## Structural patterns

### Transparent containers
Transparent containers create a modern glass look and feel. To enable transparency on a container, apply `pf-v6-theme-glass` to the `html` element class list, similar to the approach for enabling the dark mode theme. 

Within Compass layouts, elements with a transparent glass background and rounded borders should be wrapped in a `<CompassPanel>` (some Compass components do this automatically). You can adjust various options of the `<CompassPanel>` to adapt its padding and behavior to fit the context of your use case. 

> To prevent styling issues, do not nest`<CompassPanel>` components while using the glass effects.

Reviewer: ```suggestion
To prevent styling issues, do not nest `<CompassPanel>` components while using the glass effects.
```
